packages/opds-web-client/src/__tests__/state-test.ts

```diff
@@ -1,42 +1,46 @@
-jest.dontMock("../state");
-jest.dontMock("../reducers/index");
-jest.dontMock("../reducers/collection");
-jest.dontMock("../reducers/book");
+import { expect } from "chai";
+import { stub } from "sinon";
 
-let fetchCollectionAndBook;
-let createFetchCollectionAndBook = jest.genMockFunction();
-createFetchCollectionAndBook.mockImplementation((collectionUrl, bookUrl) => fetchCollectionAndBook);
 let dispatch = () => {};
 let initialState = "initial state";
 let alteredState = "state with collection and book";
 let testState = initialState;
-jest.setMock("../components/mergeRootProps", { createFetchCollectionAndBook });
-jest.setMock("../store", { default: () => { return { dispatch, getState: () => testState }; } });
+import * as mergeRootProps from "../components/mergeRootProps";
+import * as store from "../store";
 
 import buildInitialState from "../state";
 
 describe("buildInitialState", () => {
   let collectionUrl = "collection url";
   let bookUrl = "book url";
+  let fetchCollectionAndBookStub;
+  let createFetchCollectionAndBookStub;
+  let storeStub;
 
   beforeEach(() => {
-    fetchCollectionAndBook = jest.genMockFunction();
-    fetchCollectionAndBook.mockReturnValue(new Promise((resolve, reject) => {
+    fetchCollectionAndBookStub = stub().returns(new Promise((resolve, reject) => {
       testState = alteredState;
       resolve({ collectionData: null, bookData: null });
     }));
-    createFetchCollectionAndBook.mockClear();
+    createFetchCollectionAndBookStub = stub(mergeRootProps, "createFetchCollectionAndBook").returns(fetchCollectionAndBookStub);
+
+    storeStub = stub(store, "default").returns({ dispatch, getState: () => testState });
+  });
+
+  afterEach(() => {
+    createFetchCollectionAndBookStub.restore();
+    storeStub.restore();
```

Comment:
I think this is the main thing I dislike about using sinon + mocha. They aren't integrated so if you stub out a function in an existing module, the stub will persist through all the subsequent tests unless you restore the function.